daemon/com.c

```diff
@@ -573,7 +574,7 @@ set_thread_id_record(
                 return;
             }
         }
-        log_message(MTC_LOG_WARNING, "COM: thread_id %d not found in thraed_id_record_table.\n", self);
+        log_message(MTC_LOG_WARNING, "COM: thread_id %lu not found in thread_id_record_table.\n", self);
```

Comment:
Not really well defined what the formatting string should be in case of `pthread_id`. Portably it does not need to be even an integral type. Not sure if I would prefer to force a cast to the type like
```c
log_message(MTC_LOG_WARNING, "COM: thread_id %lu not found in thread_id_record_table.\n", (unsigned long) self);
```
or simply rely on compiler warnings on the formatting string.

As a minor and OT, below, in another hunk, `thread_id` is formatted as hexadecimal while there decimal is used.

Reply:
I didn't find a way to portably print `pthread_t`. Some say that there are OSes where it is a `struct` and not an integer type.
Apparently the only thing you can do with this ID is to compare it with `pthread_equal` (so the `==` comparison a few lines above is not quite right, but works on Linux), join it, set various properties of it. But looks like noone thought of debugging pthread programs :(